version-ranges/src/lib.rs

```diff
@@ -1130,6 +1211,12 @@ pub mod tests {
             }
             assert!(simp.segments.len() <= range.segments.len())
         }
+
+        #[test]
+        fn from_iter_valid(segments in proptest::collection::vec(any::<(Bound<u32>, Bound<u32>)>(), ..30)) {
+            // We check invariants in the method.
+            Ranges::from_iter(segments.clone());
```

Comment:
We could strengthen this test by checking that the result is equivalent to iteratively calling union.

Reply:
This caught a significant oversight!